### ISSUE TITLE
Add support for creating .gitignore from template

### ISSUE THREAD

**Issue by jihchi** [pikapkg/create-snowpack-app#164]
It seems like currently create-snowpack-app won't pick up `.gitignore` from template?

https://github.com/pikapkg/create-snowpack-app/blob/a795622011744fe8aee8e712b67704bac30edad8/packages/create-snowpack-app/index.js#L93-L96

Am I misunderstanding here? it would be nice if we can derive `.gitignore` from template if there is one.

**Comment by stramel** [comment]
I've ran into issues with this in the past on generators, I can take a look here soon

**Comment by FredKSchott** [comment]
Thanks @stramel ! Hopefully it's just that we're overwriting it naively. If a template provides their own, we should honor that

**Comment by stramel** [comment]
@FredKSchott `.gitignore` has always been problematic in template generators. https://github.com/npm/npm/issues/1862

The workaround that I have seen is for the template to have a `._gitignore` or something like similar that we can handle as a special case and rename it. However, if a template doesn't do this, the file still becomes `.npmignore`

**Comment by jihchi** [comment]
Just FYI. [`create-react-app`](https://create-react-app.dev/docs/custom-templates/#the-template-folder) takes `gitignore` and rename it to `.gitignore`. ([code reference](https://github.com/facebook/create-react-app/blob/2da5517689b7510ff8d8b0148ce372782cb285d7/packages/react-scripts/scripts/init.js#L264-L278))

**Comment by stramel** [comment]
@jihchi yep, I remember participating in the issue that discussed the solution. Couldn't remember what it was though.
